modules/core/src/v2/coins/sol.ts

```diff
@@ -133,15 +157,32 @@ export class Sol extends BaseCoin {
     params: SolSignTransactionOptions,
     callback?: NodeCallback<SignedTransaction>
   ): Bluebird<SignedTransaction> {
-    throw new MethodNotImplementedError('signTransaction method not implemented');
+    const self = this;
+    return co<SignedTransaction>(function* () {
+      const factory = accountLib.register(self.getChain(), accountLib.Sol.TransactionBuilderFactory);
+      const txBuilder = factory.from(params.txPrebuild.txBase64);
+      txBuilder.sign({ key: params.prv });
+      const transaction: any = yield txBuilder.build();
+
+      if (!transaction) {
+        throw new Error('Invalid transaction');
+      }
+
+      const response = {
+        txBase64: transaction.toBroadcastFormat(),
+      };
+      return response;
```

Comment:
response doesn't match any of the allowed types for `SignedTransaction`. since we're focusing on single sig, we can assume that this will always produce a fully signed transaction.

i.e.
```
return {
  txHex: transaction.toBroadcastFormat()
}
```